backend/wmg/data/load_corpus.py

```diff
@@ -177,43 +182,93 @@ def get_X_raw(anndata_object: anndata.AnnData) -> Union[np.ndarray, sparse.spmat
     return raw_expression_matrix if raw_expression_matrix is not None else anndata_object.X
 
 
-def save_raw(anndata_object: anndata.AnnData, group_name: str, global_var_index: numpy.ndarray, first_obs_idx: int):
+def transform_dataset_raw_counts_to_rankit(
+    anndata_object: anndata.AnnData, corpus_path: str, global_var_index: numpy.ndarray, first_obs_idx: int
+):
     """
-    Apply rankit normalization to raw expression values and save to the tiledb corpus object
+    Apply rankit normalization to raw count expression values and save to the tiledb corpus object
     """
-    array_name = f"{group_name}/raw"
+    array_name = f"{corpus_path}/{INTEGRATED_ARRAY_NAME}"
     expression_matrix = get_X_raw(anndata_object)
     logger.info(f"saving {array_name}...")
     stride = max(int(np.power(10, np.around(np.log10(1e9 / expression_matrix.shape[1])))), 10_000)
     with tiledb.open(array_name, mode="w") as array:
         for start in range(0, expression_matrix.shape[0], stride):
             end = min(start + stride, expression_matrix.shape[0])
-            csr_sparse_raw_expression_matrix = sparse.csr_matrix(expression_matrix[start:end, :])
-
-            coo_sparse_raw_expression_matrix = csr_sparse_raw_expression_matrix.tocoo(copy=False)
-            rows = coo_sparse_raw_expression_matrix.row + start + first_obs_idx
-            cols = global_var_index[coo_sparse_raw_expression_matrix.col]
-            raw_data = coo_sparse_raw_expression_matrix.data
-            rankit_normalized_coo_sparse_raw_expression_matrix = rankit(csr_sparse_raw_expression_matrix).tocoo(
-                copy=False
-            )
-            assert np.array_equal(
-                coo_sparse_raw_expression_matrix.row, rankit_normalized_coo_sparse_raw_expression_matrix.row
-            )
-            assert np.array_equal(
-                coo_sparse_raw_expression_matrix.col, rankit_normalized_coo_sparse_raw_expression_matrix.col
+            raw_expression_csr_matrix = sparse.csr_matrix(expression_matrix[start:end, :])
+
+            # Compute RankIt
+            rankit_integrated_csr_matrix = rankit(raw_expression_csr_matrix)
+
+            rankit_integrated_coo_matrix = filter_out_rankits_with_low_expression_counts(
+                rankit_integrated_csr_matrix, raw_expression_csr_matrix, expect_majority_filtered=True
             )
-            rankit_data = rankit_normalized_coo_sparse_raw_expression_matrix.data
 
-            array[rows, cols] = {"data": raw_data, "rankit": rankit_data}
+            global_rows = rankit_integrated_coo_matrix.row + start + first_obs_idx
+            global_cols = global_var_index[rankit_integrated_coo_matrix.col]
```

Comment:
now computing global_{rows,cols} _after_ rankit filtering is done, so that we can write out the reduced set of coords & data to the integrated corpus Array.